### DISCUSSION TITLE
Convert string "1,2,3,4" to array of ints

### DISCUSSION THREAD

**Discussion by scottee** [duckdb/duckdb#3541] [General]
I have a string column that is a comma-separated list of numbers. How do I convert it to an array of ints? I can use string_split() to turn the string to array[string], but how do I cast each array member to an int? In presto I would do something like

    `transform( split( my_str_col, ','), x -> cast(x as int))`

Is there anything equivalent in duckdb?

**Comment by hawkfish** [comment]
We are currently extending the set of operations that can be performed on `LIST`s, and something like `transform` (`map`?) is on the list.

**Comment by Mytherin** [comment]
In this case, you should just be able to cast to a list of integers:

```sql
D select split('1,2,3,4', ',')::int[] l;
┌──────────────┐
│      l       │
├──────────────┤
│ [1, 2, 3, 4] │
└──────────────┘
```
